### DISCUSSION TITLE
TMX vs. TSX name for GnuCash

### DISCUSSION THREAD

**Discussion by GitGord** [finance-quote/finance-quote#252] [General]
I see the changes to the TMX module are scheduled to be included in the next release, that's great, thanks.
There may be an issue making this module work with GnuCash. I'm not 100% clear on how it works but I understand there is a file in GnuCash (https://github.com/Gnucash/gnucash/blob/maint/libgnucash/engine/gnc-commodity.c) that it uses to match modules to names in the list of sources for quotes. I believe currently that file points to a module named TSX not TMX, so it's possible this file needs a change. As I say, I'm not 100% clear on how this works.

Gord

**Comment by bpschuck** [comment]
No changes to GnuCash necessary. The sources defined in `gnc-commodity.c` are the ones that appear in the `Single` quote source type drop down menu, it does not contain all the available modules. `tmx` does already show in the `Unknown` drop down list. There is no TSX module, that I believe is a typo from one of the other maintainers when creating the `Modules-README.yml` file.

**Comment by GitGord** [comment]
OK, I guess I misunderstood. I thought TMX.pm was the replacement for an old/removed TSX module and therefore a change would have to be made in GC to point to the new module. Anyway, I did see that the tmx source was available in the unknown section in GC but wouldn't it be better to have it appear in the 'normal' Single quote source list?

**Comment by bpschuck** [comment]
> Anyway, I did see that the tmx source was available in the unknown
> section in GC but wouldn't it be better to have it appear in the
> 'normal' Single quote source list?

That is for the GnuCash developers to decide.

**Comment by GitGord** [comment]
I apologize for not being clear in my last message. What I meant to ask is:
Do you think GnuCash should use the TMX.pm module to get quotes from the Toronto Stock Exchange? I believe the answer is yes but I'm new to this stuff and don't know all the ins and outs.
If you think the answer to the above question is yes as well, is there some way of making the GnuCash maintainers aware of this or is it appropriate to create a Pull request to change the gnc-commodity.c file? 
I'm not sure about the timing. If GC does a release with the changes, it will still not work if it happens before FQ does a release.

**Comment by bpschuck** [comment]
Your message was clear. But your question would be more for the people on the gnucash-devel email list or the GnuCash Bugzilla. While one project (GnuCash) has an optional dependency on the other (Finance::Quote), they are two separate projects with different teams. Finance::Quote is used by a couple other OpenSource finance applications as well as an unknown number of personal tools that others may have developed for themselves.

**Comment by GitGord** [comment]
Thank you for taking the time to explain this to me.
I will do as you suggest and contact the GunCash team. (and stop pestering you! LOL)
